examples/java/testcontainer/src/test/java/com/grafana/example/TestcontainerTest.java

```diff
@@ -43,25 +48,63 @@ void testExportMetric() {
     var app = new OtelApp();
     app.run();
 
+    HttpClient client = HttpClient.newHttpClient();
     String query =
         URLEncoder.encode(
-            "sold_items_total{job=\"otel-java-test\",service_name=\"otel-java-test\",tenant=\"tenant1\"}",
+            "sold_items_total{job=\"otel-java-test\",tenant=\"tenant1\"}",
             StandardCharsets.UTF_8);
     String prometheusHttpUrl = lgtm.getPromehteusHttpUrl() + "/api/v1/query?query=" + query;
 
-    HttpClient client = HttpClient.newHttpClient();
     HttpRequest request = HttpRequest.newBuilder().uri(URI.create(prometheusHttpUrl)).build();
 
-    // Total time: 18.448 s (for the whole test, will take longer when the image needs to be
-    // downloaded)
     await()
-        .atMost(Duration.ofSeconds(10))
+        .atMost(Duration.ofSeconds(20))
         .until(
             () -> {
               HttpResponse<String> response =
                   client.send(request, HttpResponse.BodyHandlers.ofString());
               String body = response.body();
               return response.statusCode() == 200 && body.contains("sold_items");
             });
+
+    // Get the Tempo datasource ID so we can query traces from grafana
+    String authHeader = "Basic " + Base64.getEncoder().encodeToString("admin:admin".getBytes(StandardCharsets.UTF_8));
```

Comment:
querying tempo directly seems to be easier - similar to https://github.com/grafana/oats/blob/main/yaml/traces.go#L14 - but using java

Reply:
I opened a pr to update the test container to open up the port to allow querying tempo directly
https://github.com/testcontainers/testcontainers-java/pull/10192

Reply:
great!

If you want to know how Loki API works: https://github.com/search?q=repo%3Agrafana%2Foats%20SearchLoki&type=code

Reply:
https://github.com/testcontainers/testcontainers-java/pull/10256